IBAnimatable/MaskDesignable.swift

```diff
@@ -50,7 +51,9 @@ public extension MaskDesignable where Self: UIView {
     case .triangle:
       maskTriangle()
     case .none:
-      layer.mask?.removeFromSuperlayer()
+      if previousMaskType != nil {
+        layer.mask?.removeFromSuperlayer()
```

Comment:
I wanted to keep the logic as simple (and understandable) as possible - so any use of `MaskType` (after initialization to `.none`) would force a `layer.mask?.removeFromSuperlayer()`. However, my intuition suggests you are right, and resetting to `.none` could reenable custom gradients.
I'll go ahead and reset previousMaskType.